File: 백준/Silver/25631. 마트료시카 합치기/마트료시카 합치기.py
import sys
input = sys.stdin.readline

T = int(input())

num = list(map(int, input().split()))

num.sort()

result = 0
compare = 1

for i in range(1, T) :
    if num[i - 1] != num[i] :
        result = max(result, compare)
        compare = 1
    else :
        compare += 1

result = max(result, compare)

print(result)

File: 백준/Silver/25631. 마트료시카 합치기/마트료시카 합치기.java
import java.io.*;
import java.util.*;

public class Main {	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		
		int N = Integer.parseInt(br.readLine());
        
		st = new StringTokenizer(br.readLine());
		int[] num = new int[N];
		for (int i = 0; i < N; i++) {
			num[i] = Integer.parseInt(st.nextToken());
		}
		Arrays.sort(num);
		
		int result = 1;
		int cnt = 1;
		for (int i = 1; i < N; i++) {
			if (num[i] == num[i-1]) {
				cnt++;
			} else {
				result = Math.max(result, cnt);
				cnt = 1;
			}
		}
		result = Math.max(result, cnt);
		
		System.out.println(result);
		br.close();
	}
}